app/src/wavevr/cpp/DeviceDelegateWaveVR.cpp

```diff
@@ -350,6 +352,26 @@ struct DeviceDelegateWaveVR::State {
         delegate->EndTouch(controller.index);
       }
       delegate->SetAxes(controller.index, immersiveAxes, kNumAxes);
+      delegate->SetHapticsCount(controller.index, 1);
+
+      uint64_t  inputFrameID = 0;
+      float pulseDuration = 0.0f, pulseIntensity = 0.0f;
+      delegate->GetHapticsFeedback(controller.index, inputFrameID, pulseDuration, pulseIntensity);
+      if (inputFrameID == 0 || pulseDuration <= 0 ) {
+        // Stopping haptics feedback.
+        if (controller.doingVibration) {
+          WVR_TriggerVibration(controller.type, WVR_InputId_Max, 0.0f, 0);
+          controller.doingVibration = false;
+        }
+      } else if (pulseIntensity > 0.0f) {
+        controller.doingVibration = true;
```

Comment:
I'm not sure what you need `doingVibration` for. Is it so you only stop it once? Why do we need to stop it since the duration is one of the parameter?

Reply:
Ok. We can remove it because the stop function actually doesn't work for Focus+.
```
WVR_TriggerVibration(WVR_DeviceType_Controller_Left, WVR_InputId_Max, 10000000, 1);
WVR_TriggerVibration(WVR_DeviceType_Controller_Left, WVR_InputId_Max, 0, 0);
```